base/src/commonMain/kotlin/monitor/ServiceMonitor.kt

```diff
@@ -33,7 +33,7 @@ abstract class DefaultServiceMonitor : ServiceMonitor {
 
     protected val TAG: String = this::class.simpleName ?: "ServiceMonitor"
 
-    protected val _isEnabled = MutableStateFlow(isServiceEnabled)
+    private val _isEnabled = MutableStateFlow(false)
```

Comment:
Thing is, old code pointed to an abstract function in its init code. That will crash if the abstract method has a reference to self (which it now has on ios to deal with freezing issues)

Also for Bluetooth you dont want this because it will initialize a CBManager which will trigger os to request permissions before it should. 

If you want a different state, overwrite it in the initialization of the subclass

Reply:
Thinking about it, it should actually be an optional boolean. It should be null if not monitoring, since if you dont monitor, you wont know the current state

Reply:
Made the default _isEnabled null for now. This should be the correct behaviour: if you dont monitor it, a flow shouldnt return a value.